packages/aws-cdk/lib/api/provisioner.ts

```diff
@@ -0,0 +1,44 @@
+import cxapi = require('@aws-cdk/cx-api');
+import { debug } from '../logging';
+import { deserializeStructure } from '../serialize';
+import { Mode } from './aws-auth/credentials';
+import { SDK } from './util/sdk';
+
+export type Template = { [key: string]: any };
+
+/**
+ * Interface for provisioners
+ *
+ * Provisioners apply templates to the cloud infrastructure.
+ */
+export interface IProvisioner {
+  readCurrentTemplate(stack: cxapi.SynthesizedStack): Promise<Template>;
+}
+
+export interface ProvisionerProps {
+  aws: SDK;
+}
+
+/**
+ * Default provisioner (applies to CloudFormation).
+ */
+export class CfnProvisioner implements IProvisioner {
```

Comment:
May I ask that we use `CloudFormation` as a prefix (I prefer we keep "Cfn" for L1 classes).